libraries/chain/include/eosio/chain/sync_call_context.hpp

```diff
@@ -24,21 +28,25 @@ class sync_call_context : public host_context {
 
    bool is_sync_call() const override { return true; }
 
-   bool is_read_only()const override;
+   bool is_read_only() const override { return read_only; };
    action_name get_sender() const override;
    void console_append(std::string_view val) override;
    void store_console_marker() override;
 
-   uint32_t               ordinal = 1;
-   action_trace&          current_action_trace;
-   account_name           sender{};
-   uint64_t               flags = 0;
-   const bool             is_caller_read_only = false;
-   std::span<const char>  data{}; // includes function name, arguments, and other information
-   std::vector<char>      return_value{};
-
-   bool has_recipient(account_name account)const override;
-   void update_db_usage( const account_name& payer, int64_t delta ) override;
+   const uint32_t               ordinal = 1;
+   action_trace&                current_action_trace;
+   const account_name           sender{};
+   const uint64_t               flags = 0;
```

Comment:
It seems like don't really even need `flags` any more in the current implementation -- until some time in the future where maybe needing more flags.

Reply:
No, I'm referring to `flags` in `sync_call_context`. afaict it's currently unnecessary and actually burdens your implementation a little since you mutate the passed flags,
https://github.com/AntelopeIO/spring/blob/fd03667f24ad0e3a8a2237c0c637a9d2523ad385/libraries/chain/host_context.cpp#L47-L51

I approved because I don't think you _need_ to get rid of it but I think it would slightly reduce quantity of code

Reply:
oh I see. I removed it as it is not needed, and we can add it back when needed as it is an implementation thing and won't break consensus.

Reply:
https://github.com/AntelopeIO/spring/pull/1410/commits/2356b8fbb63a536a5e5fca17608bbb19c4bc5007

Reply:
Thanks! I was a little hesitant to remove the flags in the constructor but that makes code even simpler. Let's do it.

Reply:
https://github.com/AntelopeIO/spring/pull/1410/commits/d7ecbe54979ba3f8507a5586ceeeff2e089c0d19